### PR TITLE
Wider iam profile

### PR DESCRIPTION
**Pull request by ChrisScottThomas**
I doubt this is correct, but it's a base

### CODE REVIEW

File: resources.tf [hunk 1]
```diff
@@ -106,3 +106,111 @@ resource "aws_s3_bucket_policy" "published_bucket_https_only" {
   bucket = aws_s3_bucket.published.id
   policy = data.aws_iam_policy_document.published_bucket_https_only.json
 }
+
+resource "aws_iam_role" "dataset_generator" {
```
Comment (mattburgess) [inline resources.tf]: `s/dataset_generator/analytical_dataset_generator/` throughout please. We'll be having a `pdm_dataset_generator` one come along shortly.

File: resources.tf [hunk 2]
```diff
@@ -106,3 +106,111 @@ resource "aws_s3_bucket_policy" "published_bucket_https_only" {
...
+}
+
+resource "aws_iam_role_policy_attachment" "ec2_for_ssm_attachment" {
+  role       = aws_iam_role.dataset_generator.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM"
+
+}
+
+#        Create and attach custom policy
+data "aws_iam_policy_document" "write_s3_dev" {
```
Comment (mattburgess) [inline resources.tf]: Can we name this `analytical_dataset_write_s3` please

Comment (mattburgess) [reply resources.tf]: Could you also add a TODO pointing to DW-3618 please? That will cover off locking this role down to defined resources.

File: resources.tf [hunk 3]
```diff
@@ -106,3 +106,111 @@ resource "aws_s3_bucket_policy" "published_bucket_https_only" {
...
+      "kms:ReEncrypt*",
+      "kms:GenerateDataKey*",
+      "kms:DescribeKey",
+    ]
+
+    # resources = [data.terraform_remote_state.security-tools.ebs_cmk.arn]
+  }
+}
+
+resource "aws_iam_policy" "write_s3_dev" {
```
Comment (mattburgess) [inline resources.tf]: Again, can we name this analytical_dataset_write_s3 please?

File: resources.tf [hunk 4]
```diff
@@ -106,3 +106,111 @@ resource "aws_s3_bucket_policy" "published_bucket_https_only" {
...
+  name        = "DatasetGeneratorWriteS3"
+  description = "Allow Dataset Generator clusters to write to S3 buckets"
+  policy      = data.aws_iam_policy_document.write_s3_dev.json
+}
+
+resource "aws_iam_role_policy_attachment" "emr_write_s3_dev" {
```
Comment (mattburgess) [inline resources.tf]: And this to `analytical_dataset_write_s3` please.

File: resources.tf (hunk 1 above)
Comment (ChrisScottThomas) [reply resources.tf]: OK I thought we were making one role to rule them all.

**Pull request merged**